File: src/main/java/org/example/kalkulationsprogramm/dto/Email/UnifiedEmailDto.java
package org.example.kalkulationsprogramm.dto.Email;

import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;

/**
 * Unified Email Response DTO für das neue Email-System.
 */
@Data
public class UnifiedEmailDto {
    private Long id;
    private String messageId;

    // Absender/Empfänger
    private String fromAddress;
    private String senderDomain;
    private String recipient;
    private String cc;

    // Inhalt
    private String subject;
    private String body;
    private String htmlBody;

    // Zeitstempel
    private LocalDateTime sentAt;
    private LocalDateTime firstViewedAt;

    @JsonProperty("isRead")
    private boolean isRead; // Computed: firstViewedAt != null

    @JsonProperty("isStarred")
    private boolean isStarred;

    // Metadaten
    private String direction; // IN, OUT
    private String zuordnungTyp; // KEINE, PROJEKT, ANFRAGE, LIEFERANT

    // Zuordnungs-Info
    private Long projektId;
    private String projektName;
    private Long anfrageId;
    private String anfrageName;
    private Long lieferantId;
    private String lieferantName;
    /** Name des erkannten Kunden (per fromAddress/recipient gegen Kunde.kundenEmails gematcht). */
    private Long kundeId;
    private String kundeName;

    // Ordner-Zuordnung (computed)
    private String folder;

    // Spam
    private Integer spamScore;

    // Thread-Informationen
    /** ID der übergeordneten E-Mail; null für Thread-Wurzeln (Root-Emails). */
    private Long parentEmailId;
    /** Anzahl direkter Antworten auf diese E-Mail. 0 = keine Antworten vorhanden. */
    private int replyCount;

    // Anhänge
    private List<AttachmentDto> attachments;
    private boolean hasAttachments;

    @Data
    public static class AttachmentDto {
        private Long id;
        private String originalFilename;
        private String mimeType;
        private Long fileSize;
        private String contentId;
        private boolean inline;
    }
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/Email/UnifiedEmailDto.kt
package org.example.kalkulationsprogramm.dto.Email

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class UnifiedEmailDto(
    var id: Long? = null,
    var messageId: String? = null,
    var fromAddress: String? = null,
    var senderDomain: String? = null,
    var recipient: String? = null,
    var cc: String? = null,
    var subject: String? = null,
    var body: String? = null,
    var htmlBody: String? = null,
    var sentAt: LocalDateTime? = null,
    var firstViewedAt: LocalDateTime? = null,
    @field:JsonProperty("isRead")
    var isRead: Boolean = false,
    @field:JsonProperty("isStarred")
    var isStarred: Boolean = false,
    var direction: String? = null,
    var zuordnungTyp: String? = null,
    var projektId: Long? = null,
    var projektName: String? = null,
    var anfrageId: Long? = null,
    var anfrageName: String? = null,
    var lieferantId: Long? = null,
    var lieferantName: String? = null,
    var kundeId: Long? = null,
    var kundeName: String? = null,
    var folder: String? = null,
    var spamScore: Int? = null,
    var parentEmailId: Long? = null,
    var replyCount: Int = 0,
    var threadLastActivityAt: LocalDateTime? = null,
    var attachments: List<AttachmentDto>? = null,
    var isHasAttachments: Boolean = false,
) {
    data class AttachmentDto(
        var id: Long? = null,
        var originalFilename: String? = null,
        var mimeType: String? = null,
        var fileSize: Long? = null,
        var contentId: String? = null,
        var isInline: Boolean = false,
    )
}
